tests/test_adjoint.py

```diff
@@ -59,6 +64,46 @@ def test_adjoint_F(self, mkey, shape, kernel, space_order):
              % (term1, term2, (term1 - term2)/term1, term1 / term2))
         assert np.isclose((term1 - term2)/term1, 0., atol=1.e-12)
 
+    @pytest.mark.parametrize('mkey, shape, space_order', [
+        # 2D tests with varying space orders
+        ('layers', (30, 35), 8), ('layers', (30, 35), 4),
+        # 3D tests with varying space orders
+        ('layers', (30, 35, 40), 8), ('layers', (30, 35, 40), 4),
+        # Constant model in 2D and 3D
+        ('constant', (30, 35), 8), ('constant', (30, 35, 40), 4),
+        ('constant', (30, 35), 8), ('constant', (30, 35, 40), 4),
```

Comment:
I would avoid to use this model. Putting anisotropy at source location is really bad for this equation so would stick to layers only.

Reply:
Is true, so the `constant-tti` model should be taken out of `demo_model` right?